modules/template/template.go

```diff
@@ -104,6 +105,15 @@ func NewFuncMap() []template.FuncMap {
 		"ThemeColorMetaTag": func() string {
 			return setting.ThemeColorMetaTag
 		},
+		"TabSizeClass": func(ec *editorconfig.Editorconfig, filename string) string {
+			if ec != nil {
+				def := ec.GetDefinitionForFilename(filename)
+				if def.TabWidth > 0 {
+					return fmt.Sprintf("tab-size-%d", def.TabWidth)
+				}
+			}
+			return "tab-size-8"
```

Comment:
Took a closer look. Default tab-width is still 8:

https://github.com/silverwind/docs/blob/patch-1/TOC.ini

It's just that the ACE editor defaults to 2-space indent on files not containing a tab character. As soon as a tab is there, it goes to 8-wide tabs indentation, overridden through `.editorconfig`.